python/cudf/cudf/core/udf/pipeline.py

```diff
@@ -199,6 +201,11 @@ def _kernel(retval, input_col_0, input_col_1, offset_0, offset_1, size):
     # Create argument list for kernel
     input_columns = ", ".join([f"input_col_{i}" for i in range(len(fr._data))])
     input_offsets = ", ".join([f"offset_{i}" for i in range(len(fr._data))])
+    extra_args = (
+        ", ".join([f"extra_arg_{i}" for i in range(len(args))]) + ","
```

Comment:
There's nothing special about the name, no. It ends up being just the name of a function argument in a python function, I think we could change `input_col_xyz`, `offset_xyz`, and `extra_arg_xyz` to anything and the code would actually work the same, as long as we changed where those args were used everywhere inside the kernel template. 

As for the trailing comma, this is to handle the empty case of`extra_args`. If we hardcode a comma then the function signature ends up getting an extra comma and causing a syntax error. It needs to be there if there are any extra args but not if there aren't. This has not come up for the other types of args because generally dataframes have at least one column, although this case I am realizing isn't tested and I'll add a check to short circuit in that case.

Reply:
Could we move the `size` parameter to the kernel defined in the kernel template above to be an earlier argument? Feel free to push if you think that's out of scope for this PR, but it feels like the underlying problem is that the "extra arguments" are not at the end of the list. It's completely valid to define a function where the argument list has a trailing comma (e.g. `f(x,)`), but the problem in this case is that you need to have the `size` after those extra args. Since the other parameters (`input_columns`, `input_offsets`) are also set when formatting the template string, `size` seems like it would make more sense as the second argument to the kernel rather than the last.

Reply:
Updated this as described